modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/TestcontainersExtension.java

```diff
@@ -160,10 +153,13 @@ boolean isDockerAvailable() {
         Optional<ExtensionContext> current = Optional.of(context);
         while (current.isPresent()) {
             ExtensionContext ctx = current.get();
-            Object testInstance = ctx.getStore(NAMESPACE).remove(TEST_INSTANCE);
-            if (testInstance != null) {
-                testInstances.add(testInstance);
+            List<Object> contextTestInstances = ctx.getTestInstances()
+                .map(TestInstances::getAllInstances)
+                .orElseGet(Collections::emptyList);
+            for (int i = contextTestInstances.size() - 1; i >= 0; --i) {
+                testInstances.add(contextTestInstances.get(i));
             }
+
             current = ctx.getParent();
         }
         return testInstances;
```

Comment:
Assuming the order is important, I think this method body can be replaced with:
```java
List<Object> allInstances = new ArrayList<>(ctx.getRequiredTestInstances().getAllInstances());
Collections.reverse(allInstances);
return new LinkedHashSet<>(allInstances);
```

Reply:
The order isn't important (sine there is no guarantee from the JVM anyways, right?), and we have `dependsOn` to enforce it.

Or is there some built-in way of ensuring the order that `TestInstances` is using?

Reply:
@britter Used LinkedHashSet (which has ordering guarantees), so I guess the order is important.

I guess the intention is to start containers of parent classes before child classes.

Reply:
I don't remember why I used a LinkedHashSet to be honest 😇